test/spmd/test_inference_spmd_dynamo_imagenet.py

```diff
@@ -0,0 +1,238 @@
+      mesh_shape = (2, num_devices // 2, 1, 1)
+      mesh = xs.Mesh(device_ids, mesh_shape, ('w', 'x', 'y', 'z'))
+      partition_spec = (0, 1, 2, 3)  # Apply sharding along all axes
+      print(
+          f'Applying sharding to convolution layers with mesh {mesh.get_logical_mesh()}'
+      )
+      for name, layer in model.named_modules():
+        if 'conv' in name:
+          xs.mark_sharding(layer.weight, mesh, partition_spec)
+    elif 'linear' in FLAGS.sharding:
+      # Shard the model's fully connected layers across addressable devices
+      mesh_shape = (num_devices, 1)
+      mesh = xs.Mesh(device_ids, mesh_shape, ('x', 'y'))
+      print(
+          f'Applying sharding to linear layers with mesh {mesh.get_logical_mesh()}'
+      )
+      partition_spec = (0, 1)
+      for name, layer in model.named_modules():
+        if 'fc' in name:
+          xs.mark_sharding(layer.weight, mesh, partition_spec)
+
+    # Input sharding
+    if 'batch' in FLAGS.sharding or 'spatial' in FLAGS.sharding:
+      if 'batch' in FLAGS.sharding and 'spatial' in FLAGS.sharding:
+        # Shard along both the batch dimension and spatial dimension
+        # If there are more than 4 devices, shard along the height axis as well
+        width_axis, height_axis = 2, num_devices // 4
+        mesh_shape = (2, 1, width_axis, height_axis)
+        input_mesh = xs.Mesh(device_ids, mesh_shape, ('B', 'C', 'W', 'H'))
+        print(
+            f'Sharding input along batch and spatial dimensions with mesh {input_mesh.get_logical_mesh()}'
+        )
+      elif 'batch' in FLAGS.sharding:
+        # Shard along batch dimension only
+        mesh_shape = (num_devices, 1, 1, 1)
+        input_mesh = xs.Mesh(device_ids, mesh_shape, ('B', 'C', 'W', 'H'))
+        print(
+            f'Sharding input along batch dimension with mesh {input_mesh.get_logical_mesh()}'
+        )
+      elif 'spatial' in FLAGS.sharding:
+        # Shard two-way along input spatial dimensions
+        mesh_shape = (1, 1, num_devices // 2, 2)
+        input_mesh = xs.Mesh(device_ids, mesh_shape, ('B', 'C', 'W', 'H'))
+        print(
+            f'Sharding input images on spatial dimensions with mesh {input_mesh.get_logical_mesh()}'
+        )
+      test_loader = pl.MpDeviceLoader(
+          test_loader,
+          device,
+          input_sharding=xs.ShardingSpec(input_mesh, (0, 1, 2, 3)))
+
+  model.eval()
+  xm.mark_step()
+  xm.wait_device_ops()
+  met.clear_all()
+
+  if FLAGS.use_dynamo:
+    print('Running torch.compile...')
+    model = torch.compile(model, backend='torchxla_trace_once')
+
+  writer = None
+  if xm.is_master_ordinal():
+    writer = test_utils.get_summary_writer(FLAGS.logdir)
+
+  def inference_loop_fn(loader):
+    start_0 = time.time()
+    for step, (data, _) in enumerate(loader):
+      if step == 1:
+        start_1 = time.time()
+      output = model(data)
+    end = time.time()
+    return end - start_0, end - start_1
```

Comment:
If this test is inference only, we should surface that in the test name.

Reply:
Where should we add a reference to inference? The main function in this file is `inference_imagenet` and the file name is `test/spmd/test_inference_spmd_dynamo_imagenet.py`

Reply:
oh I miss the inference in the name, my bad.